### DISCUSSION TITLE
How to use service inside a another service

### DISCUSSION THREAD

**Discussion by arijit04** [loopbackio/loopback-next#7347] [Q&A] [answered]
I have created  2 services. UserService and RoleService. Now from UserService I want to call RoleService functions. So in UserService constructor I am using @service. 
Like: 
`constructor(
    @repository(UserRepository) private userRepository: UserRepository,
    @service(RoleService) private roleService: RoleService,
  ) { }`
  
  It is booting up my api server but when I am calling any api end point it is throwing error. 
  can some one help me? 
  
  Thanks in advance.

**Comment by raymondfeng** [comment, marked as answer]
What error do you see?

**Reply by arijit04** [reply]
Unhandled error in POST /login: 500 Error: No design-time type metadata found while inspecting UserService.constructor[0]. You can either use `@service(ServiceClass)` or ensure `emitDecoratorMetadata` is enabled in your TypeScript configuration. Run `tsc --showConfig` to print the final TypeScript configuration of your project.

**Reply by raymondfeng** [reply]
Do you have circular ref of `RoleService`? If you can show the code of your services, it may be easier to tell.

**Reply by arijit04** [reply]
No there no circular ref.

**Reply by raymondfeng** [reply]
Circular refs can be created or `import` or `require`. See https://nodejs.org/api/modules.html#modules_cycles.

**Reply by raymondfeng** [reply]
That's just one of the possible causes I would like to rule out. Try `npx madge --circular <your-app-main.js>` to check.

**Reply by arijit04** [reply]
Yes Circular refs is present. :(

**Reply by raymondfeng** [reply]
Some of the such refs can be fixed by importing the leaf file directly.

**Reply by arijit04** [reply]
Thank you so much. It is working perfectly. 👍 :)
